framework/src/meshgenerators/ParsedGenerateSideset.C

```diff
@@ -122,6 +130,27 @@ ParsedGenerateSideset::generate()
       const std::vector<Point> & normals = _fe_face->get_normals();
       _fe_face->reinit(elem, side);
 
+      // check if the neighboring elems subdomain is included
+      if (_check_neighbor_subdomains)
+      {
+        const Elem * neighbor = elem->neighbor_ptr(side);
+        // if the neighbor does not exist, then skip this face; we only add sidesets
+        // between existing elems if _check_neighbor_subdomains is true
+        if (!neighbor)
+          continue;
+
+        subdomain_id_type curr_neighbor_subdomain = neighbor->subdomain_id();
+        if (std::find(_included_neighbor_ids.begin(),
+                      _included_neighbor_ids.end(),
+                      curr_neighbor_subdomain) == _included_neighbor_ids.end())
+          continue;
+      }
+
+      if (_check_neighbor_subdomains &&
+          std::find(_included_neighbor_ids.begin(), _included_neighbor_ids.end(), curr_subdomain) ==
+              _included_ids.end())
+        continue;
```

Comment:
Why is this here?

Reply:
Oh, yeah.... and here's your error too on line 151: comparing _included_neighbors_ids and _included_ids.

I think this can just be removed?